core/services/ocr2/plugins/llo/integration_test.go

```diff
@@ -0,0 +1,395 @@
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+	"go.uber.org/zap/zaptest/observer"
+
+	chainselectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+	datastreamsllo "github.com/smartcontractkit/chainlink-data-streams/llo"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/confighelper"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3confighelper"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/assets"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/llo-feeds/generated/channel_config_store"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/llo-feeds/generated/channel_verifier"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/llo-feeds/generated/verifier_proxy"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/csakey"
+	"github.com/smartcontractkit/chainlink/v2/core/services/llo"
+	lloevm "github.com/smartcontractkit/chainlink/v2/core/services/llo/evm"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm"
+)
+
+var (
+	fNodes           = uint8(1)
+	nNodes           = 4 // number of nodes (not including bootstrap)
+	multiplier int64 = 100000000
+)
+
+func setupBlockchain(t *testing.T) (*bind.TransactOpts, *backends.SimulatedBackend, *channel_verifier.ChannelVerifier, common.Address, *channel_config_store.ChannelConfigStore, common.Address) {
+	steve := testutils.MustNewSimTransactor(t) // config contract deployer and owner
+	genesisData := core.GenesisAlloc{steve.From: {Balance: assets.Ether(1000).ToInt()}}
+	backend := cltest.NewSimulatedBackend(t, genesisData, uint32(ethconfig.Defaults.Miner.GasCeil))
+	backend.Commit()                                  // ensure starting block number at least 1
+	stopMining := cltest.Mine(backend, 1*time.Second) // Should be greater than deltaRound since we cannot access old blocks on simulated blockchain
+	t.Cleanup(stopMining)
+
+	// Deploy contracts
+	verifierProxyAddr, _, _, err := verifier_proxy.DeployVerifierProxy(steve, backend, common.Address{}) // zero address for access controller disables access control
+	require.NoError(t, err)
+
+	verifierAddress, _, verifierContract, err := channel_verifier.DeployChannelVerifier(steve, backend, verifierProxyAddr)
+	require.NoError(t, err)
+	configStoreAddress, _, configStoreContract, err := channel_config_store.DeployChannelConfigStore(steve, backend)
+	require.NoError(t, err)
+
+	backend.Commit()
+
+	return steve, backend, verifierContract, verifierAddress, configStoreContract, configStoreAddress
+}
+
+func detectPanicLogs(t *testing.T, logObservers []*observer.ObservedLogs) {
```

Comment:
Would it be possible to observe at `DPanicLevel` instead of filtering afterwards?

Reply:
Might be, there should probably be a more broadly available test helper for this

Reply:
What are you trying to achieve exactly?

It seems hacky to be printing a non-standard format. Why not just emit and observe at the level you want in the first place?

Reply:
Nothing, this was a copypasta. It has been removed